Install is broken with node 20

If I try to build it with node 20 then I get this error:
```
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from ../src/main.cpp:3:
npm ERR! In file included from ../../nan/nan.h:180:
npm ERR! ../../nan/nan_callbacks.h:55:23: error: no member named 'AccessorSignature' in namespace 'v8'
npm ERR! typedef v8::Local<v8::AccessorSignature> Sig;
npm ERR!                   ~~~~^
npm ERR! In file included from ../src/main.cpp:3:
npm ERR! ../../nan/nan.h:686:39: warning: 'IdleNotificationDeadline' is deprecated: Use MemoryPressureNotification() to influence the GC schedule. [-Wdeprecated-declarations]
npm ERR!     return v8::Isolate::GetCurrent()->IdleNotificationDeadline(
npm ERR!                                       ^
npm ERR! ~/Library/Caches/node-gyp/20.5.1/include/node/v8-isolate.h:1291:3: note: 'IdleNotificationDeadline' has been explicitly marked deprecated here
npm ERR!   V8_DEPRECATE_SOON(
npm ERR!   ^
npm ERR! ~/Library/Caches/node-gyp/20.5.1/include/node/v8config.h:550:39: note: expanded from macro 'V8_DEPRECATE_SOON'
npm ERR! # define V8_DEPRECATE_SOON(message) [[deprecated(message)]]
npm ERR!                                       ^
npm ERR! 1 warning and 1 error generated.
npm ERR! make: *** [Release/obj.target/diskusage/src/main.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (~/.nvm/versions/node/v20.5.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:203:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm ERR! gyp ERR! System Darwin 22.5.0
npm ERR! gyp ERR! command "./.nvm/versions/node/v20.5.1/bin/node" "./.nvm/versions/node/v20.5.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd ./node_modules/diskusage
npm ERR! gyp ERR! node -v v20.5.1
npm ERR! gyp ERR! node-gyp -v v9.4.0
npm ERR! gyp ERR! not ok
```

--- a/fallback.js
+++ b/fallback.js
@@ -1,0 +1,29 @@
+var native = require("./build/Release/diskusage.node");
+var promise = typeof Promise !== "undefined" ? Promise : require("es6-promise").Promise;
+
+exports.check = function(path, callback) {
+  if (callback) {
+    return check(path, callback);
+  }
+
+  return new promise(function (resolve, reject) {
+    check(path, function (err, result) {
+      err ? reject(err) : resolve(result);
+    });
+  });
+};
+
+exports.checkSync = native.getDiskUsage;
+
+function check(path, callback) {
+  var result = undefined;
+  var error = undefined;
+
+  try {
+    result = native.getDiskUsage(path);
+  } catch (error_) {
+    error = error_;
+  }
+
+  callback(error, result);
+}

--- a/index.js
+++ b/index.js
@@ -1,29 +1,56 @@
-var native = require("./build/Release/diskusage.node");
-var promise = typeof Promise !== "undefined" ? Promise : require("es6-promise").Promise;
+var fs = require('fs');
 
-exports.check = function(path, callback) {
-  if (callback) {
-    return check(path, callback);
-  }
+// Detect presence of fs.statfs
+if(typeof fs.statfs === 'function') {
+  exports.check = function(path, callback) {
+    if (callback) {
+      return check(path, callback);
+    }
 
-  return new promise(function (resolve, reject) {
-    check(path, function (err, result) {
-      err ? reject(err) : resolve(result);
+    return new Promise(function (resolve, reject) {
+      check(path, function (err, result) {
+        err ? reject(err) : resolve(result);
+      });
     });
-  });
-};
-
-exports.checkSync = native.getDiskUsage;
-
-function check(path, callback) {
-  var result = undefined;
-  var error = undefined;
-
-  try {
-    result = native.getDiskUsage(path);
-  } catch (error_) {
-    error = error_;
   }
 
-  callback(error, result);
+  function check(path, callback) {
+    fs.statfs(path, function (err, result_) {
+      var result = undefined;
+
+      if (!err) {
+        result = {
+          available: result_.bavail * result_.bsize,
+          free: result_.bfree * result_.bsize,
+          total: result_.blocks * result_.bsize
+        };
+      }
+
+      callback(err, result);
+    });
+  }
+
+  exports.checkSync = function(path, callback) {
+    var result = undefined;
+    var error = undefined;
+  
+    try {
+      var result_ = fs.statfsSync(path);
+
+      result = {
+        available: result_.bavail * result_.bsize,
+        free: result_.bfree * result_.bsize,
+        total: result_.blocks * result_.bsize
+      }
+    } catch(error_) {
+      error = error_;
+    }
+
+    callback(error, result);
+  }
+} else {
+  var diskusage = require('./fallback.js');
+
+  exports.check = diskusage.check;
+  exports.checkSync = diskusage.checkSync;
 }

--- a/index.js
+++ b/index.js
@@ -30,23 +30,14 @@ if(typeof fs.statfs === 'function') {
     });
   }
 
-  exports.checkSync = function(path, callback) {
-    var result = undefined;
-    var error = undefined;
-  
-    try {
-      var result_ = fs.statfsSync(path);
+  exports.checkSync = function(path) {
+    var result = fs.statfsSync(path);
 
-      result = {
-        available: result_.bavail * result_.bsize,
-        free: result_.bfree * result_.bsize,
-        total: result_.blocks * result_.bsize
-      }
-    } catch(error_) {
-      error = error_;
+    return {
+      available: result.bavail * result.bsize,
+      free: result.bfree * result.bsize,
+      total: result.blocks * result.bsize
     }
-
-    callback(error, result);
   }
 } else {
   var diskusage = require('./fallback.js');

--- a/package.json
+++ b/package.json
@@ -6,11 +6,11 @@
   "typings": "index.d.ts",
   "gypfile": true,
   "dependencies": {
-    "es6-promise": "^4.2.5",
-    "nan": "^2.14.0"
+    "es6-promise": "^4.2.8",
+    "nan": "^2.18.0"
   },
   "devDependencies": {
-    "eslint": "^6.0.1"
+    "eslint": "^8.49.0"
   },
   "repository": {
     "type": "git",

